test_pull_requests

```diff
@@ -935,6 +1010,119 @@ popd
+
+      valid_comment_exists = false
+      comments.each do |comment|
+        # we want to allow users to link to issues using full URLs or the
+        # short-hand #1234 GitHub links
+        comment_contains_issue_ref = comment['body'] =~ general_issue_spec || comment['body'] =~ general_ref_spec
+        # we can't allow the bot to be a valid author, or we would pick up
+        # comments where the bot explains the flake syntax to GitHub users
+        comment_author_valid = comment['user']['login'] != Properties['bot_github_user']
+
+        if comment_contains_issue_ref && comment_author_valid
+          $stderr.puts "  Investigating comment by #{comment['user']['login']}: #{comment['html_url']}"
+          # if we find a comment that matches the flake job syntax from someone
+          # trusted, we need to validate that the comment's issue list contains
+          # at least one flake issue
+          comment['body'].scan(general_issue_spec) do |issue|
+            $stderr.puts "      Determining if issue #{issue[0]} meets criteria..."
+            if issue[0] =~ flake_issue_spec
+              issue_id = $1.to_i
+              if issue_has_label?(issue_id, flake_config['repo'], flake_config['label'])
+                $stderr.puts "      Issue #{issue[0]} is a valid flake issue"
+                valid_comment_exists = true
+                break
+              end
+            end
+          end
+
+          comment['body'].scan(general_ref_spec) do |reference|
+            $stderr.puts "      Determining if reference ##{reference[0]} meets criteria..."
+            # GitHub issue references will always link to the same org/repo that
+            # the comment containing the reference is posted in, so we only need
+            # to validate that the ref is to an issue with the correct label
+            issue_id = reference[0].to_i
+            if issue_has_label?(issue_id, flake_config['repo'], flake_config['label'])
+              $stderr.puts "      Reference ##{reference[0]} points to a valid flake issue"
+              valid_comment_exists = true
+              break
+            end
+          end
+
+          break if valid_comment_exists
+        end
+      end
+
+      valid_comment_exists
+    end
+
+    # issue_has_label? determines if the issue at the
+    # repo and issue ID has a label with the given name
+    def issue_has_label?(issue_id, repo, label_name)
+      get_labels(issue_id, repo).each do |label|
+        if label["name"] == label_name
+          return true
+        end
+      end
+      return false
+    end
+
+    # format_teams builds a list of links to team rosters from a list of team IDs
+    def format_teams(team_ids)
+      links = []
+      team_ids.each do |id|
+        url, name = team_url_and_name(id)
+        links << "[#{name}](#{url})"
+      end
+      if links.length == 1
+        %{the #{links[0]} group}
```

Comment:
Because the rest of the world hasn't adopted %{} as the universal mechanism for a string, I would prefer if you stuck with " unless it's a multiple line string or a string full of "s